src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -499,7 +499,9 @@ private void Get_CORINFO_SIG_INFO(MethodDesc method, out CORINFO_SIG_INFO sig, b
             Get_CORINFO_SIG_INFO(method.Signature, out sig);
 
             // Does the method have a hidden parameter?
-            if (method.RequiresInstArg() && !isFatFunctionPointer && !_compilation.TypeSystemContext.IsSpecialUnboxingThunkTargetMethod(method))
+            ArrayMethod arrayMethod = method as ArrayMethod;
+            if ((method.RequiresInstArg() && !isFatFunctionPointer && !_compilation.TypeSystemContext.IsSpecialUnboxingThunkTargetMethod(method))
+                || (arrayMethod != null && arrayMethod.Kind == ArrayMethodKind.Address))
```

Comment:
Something in here is really begging for a helper function that makes this conditional simpler, and clearer.